Sources/Basics/Sandbox.swift

```diff
@@ -0,0 +1,88 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import TSCBasic
+import TSCUtility
+
+public enum Sandbox {
+    public static func apply(
+        command: [String],
+        writableDirectories: [AbsolutePath] = [],
+        strictness: Strictness = .default
+    ) -> [String] {
+        #if os(macOS)
+        let profile = macOSSandboxProfile(writableDirectories: writableDirectories, strictness: strictness)
+        return ["/usr/bin/sandbox-exec", "-p", profile] + command
+        #else
+        // rdar://40235432, rdar://75636874 tracks implementing sandboxes for other platforms.
+        return command
+        #endif
+    }
+
+    public enum Strictness: Equatable {
+        case `default`
+        case manifest_pre_53 // backwards compatibility for manifests
+    }
+}
+
+// MARK: - macOS
+
+#if os(macOS)
+fileprivate func macOSSandboxProfile(
+    writableDirectories: [AbsolutePath],
+    strictness: Sandbox.Strictness
+) -> String {
+    var contents = "(version 1)\n"
+
+    // Deny everything by default.
+    contents += "(deny default)\n"
+
+    // Import the system sandbox profile.
+    contents += "(import \"system.sb\")\n"
+
+    // Allow reading all files; ideally we'd only allow the package directory and any dependencies,
+    // but all kinds of system locations need to be accessible.
+    contents += "(allow file-read*)\n"
+
+    // This is needed to launch any processes.
+    contents += "(allow process*)\n"
```

Comment:
It would be nice to find a way to restrict shelling out, if we wanted to.  But when I do that then the compiled manifest itself can't run.  Maybe it's possible to allow `sandbox-exec` to shell out to just the compiled manifest but not allow further shelling out.